basiclti/tsugi-util/src/java/org/tsugi/basiclti/BasicLTIUtil.java

```diff
@@ -1336,6 +1336,24 @@ public static Date parseIMS8601(String timestamp) {
 		return null;
 	}
 
+	/**
+	 * Shift a date from the JVM Timezone to Another Timezone
+	 *
+	 * Usually the resulting date is then usually moved into UTC for transport
+	 */
+	public static Date shiftJVMDateToTimeZone(Date date, String timeZone) {
+		if ( date == null || timeZone == null ) return null;
+
+		// Get the JVM TimeZone
+		TimeZone tzJVM = TimeZone.getDefault();
+		TimeZone tzNew = TimeZone.getTimeZone(timeZone);
+		long dueTime = date.getTime();
+		// Shift into UTC and then back to the destination timezone
+		dueTime = dueTime - tzJVM.getRawOffset() + tzNew.getRawOffset();
```

Comment:
yuck!

maybe this is low-stakes code (e.g., the exact time isn't super important). but if precision is important, this should really be done by the experts:

https://www.baeldung.com/java-date-to-localdate-and-localdatetime

Reply:
Sam - In regards to the accuracy, these are due *dates* in Sakai - which are not even date time values let alone millisecond values.  Just date.  

I read the site you mention and none of the code in the site you reference talks about _**shifting**_ into a destination timezone different that the current server's time zone.  Which is the only problem that I need to solve here.

Reply:
ZonedDateTime ldtZoned = ldt.atZone(ZoneId.systemDefault());

ZonedDateTime utcZoned = ldtZoned.withZoneSameInstant(ZoneId.of("UTC"));

Reply:
The ZonedDateTime really is obsessed with changing what is printed and shown.  I tried code like:

`ZonedDateTime ldtZoned = ZonedDateTime.ofInstant(date.toInstant(), ZoneId.systemDefault());
 ZonedDateTime shiftZoned = ldtZoned.withZoneSameLocal(ZoneId.of(timeZone));
 retval = java.util.Date.from(shiftZoned.toInstant());`

And also with `withZoneSameInstant()` and they did not actually shift after being converted back in a JVM Date and Converted to ISO8601 /l UTC.  

I tried to use the fancy code :)